File: wordNet/src/main/java/org/sqlunet/wordnet/sql/SynsetsQueryFromWordId.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.wordnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

/**
 * Query for synsets containing a given word
 *
 * @author Bernard
 */
class SynsetsQueryFromWordId extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.SynsetsQueryFromWordId; // ;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param wordId     target word id
	 */
	@SuppressWarnings("boxing")
	public SynsetsQueryFromWordId(final SQLiteDatabase connection, final long wordId)
	{
		super(connection, SynsetsQueryFromWordId.QUERY);
		setParams(wordId);
	}

	/**
	 * Get synset id
	 *
	 * @return synset id
	 */
	public long getSynsetId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get synset definition
	 *
	 * @return definition
	 */
	public String getDefinition()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get synset pos id
	 *
	 * @return pos
	 */
	public String getPos()
	{
		assert this.cursor != null;
		return this.cursor.getString(2);
	}

	/**
	 * Get synset domain id
	 *
	 * @return synset domain id
	 */
	public int getDomainId()
	{
		assert this.cursor != null;
		return this.cursor.getInt(3);
	}

	/**
	 * Get sample data
	 *
	 * @return samples as a semicolon-separated string
	 */
	public String getSample()
	{
		assert this.cursor != null;
		return this.cursor.getString(4);
	}
}

File: wordNet/src/main/java/org/sqlunet/wordnet/sql/SynsetsQueryFromWordId.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.wordnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery

/**
 * Query for synsets containing a given word
 *
 * @param connection connection
 * @param wordId     target word id
 *
 * @author Bernard
 */
class SynsetsQueryFromWordId(connection: SQLiteDatabase, wordId: Long) : DBQuery(connection!!, QUERY) {

    init {
        setParams(wordId)
    }

    /**
     * Synset id
     */
    val synsetId: Long
        get() = cursor!!.getLong(0)

    /**
     * Synset definition
     */
    val definition: String
        get() = cursor!!.getString(1)

    /**
     * Synset pos id
     */
    val pos: String
        get() = cursor!!.getString(2)

    /**
     * Synset domain id
     */
    val domainId: Int
        get() = cursor!!.getInt(3)

    /**
     * Samples as a semicolon-separated string
     */
    val sample: String
        get() = cursor!!.getString(4)

    companion object {
        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.SynsetsQueryFromWordId // 
    }
}